core/src/main/java/me/mykindos/betterpvp/core/combat/throwables/ThrowableItem.java

```diff
@@ -43,6 +45,14 @@ public ThrowableItem(Item item, LivingEntity thrower, String name, long expireTi
         immune = new ArrayList<>();
     }
 
+    public void setChargePercent(int chargePercent) {
```

Comment:
I don't understand why you are doing this here, and not keeping it contained in FleshHook. Whenever you shoot a fleshhook, you should be able to keep track of it in a WeakHashmap, and on checking for collision you check to see if that fleshhook is in your list.